modules/workload-identity/README.md

```diff
@@ -0,0 +1,86 @@
+# terraform-google-workload-identity
+
+[`Workload Identity`](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity) is the recommended way to access GCP services from Kubernetes.
+
+This module creates:
+
+* GCP Service Account
+* IAM Service Account binding to `roles/iam.workloadIdentityUser`
+* Optionally, a Kubernetes Service Account
+
+## Usage
+
+The `terraform-google-workload-identity` can create a kubernetes service account for you, or use an existing kubernetes service account.
+
+### Creating a Workload Identity
+
+```hcl
+module "my-app-workload-identity" {
+  source    = "terraform-google-modules/terraform-google-kubernetes-engine/modules/workload-identity"
+  name      = "my-application-name"
+  namespace = "default"
+  project   = "my-gcp-project-name"
+}
+```
+
+This will create:
+
+* GCP Service Account named: `my-application-name@my-gcp-project-name.iam.gserviceaccount.com`
+* Kubernetes Service Account named: `my-application-name` in the `default` namespace
+* IAM Binding (`roles/iam.workloadIdentityUser`) between the service accounts
+
+Usage from a kubernetes deployment:
+
+```yaml
+metadata:
+  namespace: default
+  # ...
+spec:
+  # ...
+  template:
+    spec:
+      serviceAccountName: my-application-name
+```
+
+### Using an existing Kubernetes Service Account
+
+An existing kubernetes service account can optionally be used. When using an existing k8s servicea account the annotation `"iam.gke.io/gcp-service-account"` must be set.
+
+```hcl
+resource "kubernetes_service_account" "preexisting" {
+  metadata {
+    name = "preexisting-sa"
+    namespace = "prod"
+    annotations = {
+      "iam.gke.io/gcp-service-account" = "preexisting-sa@${var.project}.iam.gserviceaccount.com"
```

Comment:
Can we make this reference an output from the module?

Reply:
The project part? This example is showing how to use a pre-existing K8s service account with workload identity. A service account *not* created with this module.

Reply:
Right, but I'd assume we could still use an output from the module?

Reply:
Updated, I think I see what you're saying.

Reply:
Actually, I think the original format was fine. This will create a cycle.

Reply:
Reverting